microsoft-edge/web-platform/ad-selection-api.md

```diff
@@ -0,0 +1,266 @@
+                        "shared_storage_api": {
+                            "ServiceNotUsedForIdentifyingUserAcrossSites": true/false
+                        },
+                        "private_aggregation_api": {
+                            "ServiceNotUsedForIdentifyingUserAcrossSites": true/false
+                        },
+                        "ad_selection_api": {
+                            "ServiceNotUsedForIdentifyingUserAcrossSites": true/false
+                        }
+                    }
+                },
+                {
+                    "platform": "android",
+                    "attestations": {}
+                }
+            ]
+        }
+    ]
+}
+```
+
+
+<!-- ---------- -->
+###### Important fields and values
+
+* `"platform":` must be `"edge"` or `"android"`.
+
+* Valid members of `"attestations":` are:
+   * `"attribution_reporting_api":`
+   * `"shared_storage_api":`
+   * `"private_aggregation_api":`
+   * `"ad_selection_api":`
+
+   Each of these `"attestations":` entries must have a single field, `"ServiceNotUsedForIdentifyingUserAcrossSites":`, with either a `true` or `false` value.  `true` means that this service is not used for identifying the user across sites.  `false` means that this service is used for identifying the user across sites.
+
+* `"ownership_token":` is the OT token generated for your individual domain registration.
+
+
+<!-- ====================================================================== -->
+## Deploy the Ad Selection API services
+
+The Ad Selection API uses a trusted execution environment (TEE) to provide a level of assurance for data integrity, data confidentiality, and code integrity; see [Confidential Computing: Hardware-Based Trusted Execution for Applications and Data](https://confidentialcomputing.io/wp-content/uploads/sites/85/2021/03/confidentialcomputing_outreach_whitepaper-8-5x11-1.pdf).  Services provided by the Ad Selection API must run in a TEE to secure the data used by these services.
+
+Ad Selection Services running in a TEE should be deployed on a cloud platform that supports the necessary security features.  Initially, services can be deployed in Azure using [Confidential ACI containers](/azure/container-instances/container-instances-confidential-overview).
+
+The Ad Selection API provides different services that need to be deployed by sellers or buyers.
+
+
+<!-- ------------------------------ -->
+#### Images for deploying services
+
+Add your User-Defined Functions within the provided images of services.  The User-Defined Functions can run custom business logic.  The User-Defined Functions run in private containers within the deployed services.
+
+Each image defines one service, and contains starter code, as functions with empty bodies, that are the relevant User-Defined Functions for that service.  Fill in the bodies of the User-Defined Functions in these images with your own custom code.
+<!-- todo: review the above added paragraphs -->
```

Comment:
Review the added intro for Images.

Reply:
Yes, it's correct. However, some images don't have UDF and have only predefine code.

Reply:
result:
Each image defines one service.  Some images don't contain UDFs, and consist entirely of predefined code.  Some images contain UDFs, as functions with an empty body or starter code, that are the relevant User-Defined Functions for that service.  Fill in the bodies of the User-Defined Functions in these images with your own custom code.